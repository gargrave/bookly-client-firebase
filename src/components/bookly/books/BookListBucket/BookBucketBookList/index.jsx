// @flow
import React from 'react';
import { array, func, shape, string } from 'prop-types';

import type { Book, BookBucket } from '../../../../../globals/flowtypes/';

import { buildClasses } from '../../../../../utils/cssHelpers';

import BookListDetail from '../../BookListDetail/';

type Props = {
  bucket: BookBucket,
  onBookClick: Function,
};

const booksFromBucket = (
  bucket: BookBucket,
  onBookClick: Function,
) => {
  return bucket.books.map((book: Book) => {
    return (
      <BookListDetail
        book={book}
        key={book.id}
        onClick={onBookClick.bind(null, book.id)}
        showAuthor={false}
      />
    );
  });
};

const BookBucketBookList = ({
  bucket,
  onBookClick,
}: Props) => {
  return (
    <div className={buildClasses(['book-bucket__book-list'])}>
      {booksFromBucket(bucket, onBookClick)}
    </div>
  );
};

BookBucketBookList.propTypes = {
  bucket: shape({
    author: string,
    books: array,
  }).isRequired,
  onBookClick: func.isRequired,
};

export default BookBucketBookList;
