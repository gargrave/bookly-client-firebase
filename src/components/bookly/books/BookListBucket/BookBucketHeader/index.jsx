// @flow
import React from 'react';
import { array, shape, string } from 'prop-types';

import type { BookBucket } from '../../../../../constants/flowtypes/';

import { buildClasses } from '../../../../../utils/cssHelpers';

type Props = {
  bucket: BookBucket,
};

function BookBucketHeader({
  bucket,
}: Props) {
  return (
    <div className={buildClasses(['book-bucket__header'])}>
      <p>{bucket.author}</p>
    </div>
  );
}

BookBucketHeader.propTypes = {
  bucket: shape({
    author: string,
    books: array,
  }).isRequired,
};

export default BookBucketHeader;
