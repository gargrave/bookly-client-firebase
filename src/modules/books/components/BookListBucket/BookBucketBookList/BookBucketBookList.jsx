// @flow
import React from 'react'
import { array, func, shape, string } from 'prop-types'

import type { Book, BookBucket } from '../../../flowtypes'

import BookListDetail from '../../BookListDetail/BookListDetail'

type Props = {
  bucket: BookBucket,
  onBookClick: Function,
}

const booksFromBucket = (bucket: BookBucket, onBookClick: Function) =>
  bucket.books.map((book: Book) => (
    <BookListDetail
      book={book}
      key={book.id}
      onClick={() => onBookClick(book.id)}
      showAuthor={false}
    />
  ))

const BookBucketBookList = ({ bucket, onBookClick }: Props) => (
  <div>{booksFromBucket(bucket, onBookClick)}</div>
)

BookBucketBookList.propTypes = {
  bucket: shape({
    author: string,
    books: array,
  }).isRequired,
  onBookClick: func.isRequired,
}

export default BookBucketBookList
