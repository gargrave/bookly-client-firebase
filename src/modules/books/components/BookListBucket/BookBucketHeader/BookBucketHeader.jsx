// @flow
import React from 'react'
import { array, bool, func, shape, string } from 'prop-types'
import styled from 'react-emotion'

import type { BookBucket } from '../../../flowtypes'

type Props = {
  bucket: BookBucket,
  expanded: boolean,
  onToggleExpanded: (event: any) => void,
}

export const StyledWrapper = styled('div')`
  align-items: center;
  display: flex;
  justify-content: space-between;
  padding-left: 4px;
  padding-right: 4px;
`

export const StyledAuthorName = styled('span')`
  color: #777;
  font-size: 1.4rem;
  margin-bottom: 2px;
  text-align: left;

  @media only screen and (max-width: 639px) {
    font-size: 1.2rem;
  }
`

export const StyledBookCount = styled('span')`
  font-size: 1rem;
  margin-left: 4px;
`

export const StyledToggle = styled('span')`
  cursor: pointer;
  font-size: 0.9rem;
  margin-left: 8px;
`

const BookBucketHeader = ({ bucket, expanded, onToggleExpanded }: Props) => {
  return (
    <StyledWrapper>
      <StyledAuthorName>
        {bucket.author}
        <StyledBookCount>{`(${bucket.books.length})`}</StyledBookCount>
      </StyledAuthorName>
      <StyledToggle onClick={onToggleExpanded}>
        {expanded ? 'collapse' : 'expand'}
      </StyledToggle>
    </StyledWrapper>
  )
}

BookBucketHeader.propTypes = {
  bucket: shape({
    author: string,
    books: array,
  }).isRequired,
  expanded: bool.isRequired,
  onToggleExpanded: func.isRequired,
}

export default BookBucketHeader
