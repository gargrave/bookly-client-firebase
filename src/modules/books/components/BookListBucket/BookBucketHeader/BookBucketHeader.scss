.bookBucketHeader {
  color: #777;
  font-size: 1.4rem;
  margin-bottom: 2px;
  text-align: left;
}

@media only screen and (max-width: 639px) {
  .bookBucketHeader {
    font-size: 1.2rem;
  }
}
