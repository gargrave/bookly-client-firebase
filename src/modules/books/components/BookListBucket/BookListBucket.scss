.bookBucket + .bookBucket {
  margin-top: 25px;
}

@media only screen and (max-width: 639px) {
  .bookBucket + .bookBucket {
    margin-top: 35px;
  }
}
