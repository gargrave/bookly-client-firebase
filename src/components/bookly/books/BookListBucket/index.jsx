// @flow
import React from 'react';
import { array, func, shape, string } from 'prop-types';

import type { BookBucket } from '../../../../globals/flowtypes/';

import { buildClasses } from '../../../../utils/cssHelpers';

import BookBucketBookList from './BookBucketBookList/';
import BookBucketHeader from './BookBucketHeader/';

import './styles.css';

type Props = {
  bucket: BookBucket,
  onBookClick: Function,
};

const BookListBucket = ({
  bucket,
  onBookClick,
}: Props) => {
  return (
    <div className={buildClasses(['book-bucket'])}>
      <BookBucketHeader
        bucket={bucket}
      />
      <BookBucketBookList
        bucket={bucket}
        onBookClick={onBookClick}
      />
    </div>
  );
};

BookListBucket.propTypes = {
  bucket: shape({
    author: string,
    books: array,
  }).isRequired,
  onBookClick: func.isRequired,
};

export default BookListBucket;
