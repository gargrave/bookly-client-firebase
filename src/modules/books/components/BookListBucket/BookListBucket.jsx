// @flow
import React from 'react'
import { func, object } from 'prop-types'
import styled from 'react-emotion'

import type { BookBucket } from '../../flowtypes'

import BookBucketBookList from './BookBucketBookList/BookBucketBookList'
import BookBucketHeader from './BookBucketHeader/BookBucketHeader'

const Styled = styled('div')`
  & + & {
    margin-top: 25px;
  }

  @media only screen and (max-width: 639px) {
    & + & {
      margin-top: 35px;
    }
  }
`
type Props = {
  bucket: BookBucket,
  onBookClick: (bookId?: string | number) => void,
}

type State = {
  expanded: boolean,
}

export default class BookListBucket extends React.Component<Props, State> {
  static propTypes = {
    bucket: object.isRequired,
    onBookClick: func.isRequired,
  }

  state = {
    expanded: true,
  }

  handleToggleExpanded = () => {
    this.setState(({ expanded }) => ({
      expanded: !expanded,
    }))
  }

  render() {
    const { bucket, onBookClick } = this.props
    const { expanded } = this.state

    return (
      <Styled>
        <BookBucketHeader
          bucket={bucket}
          expanded={expanded}
          onToggleExpanded={this.handleToggleExpanded}
        />
        {expanded && (
          <BookBucketBookList bucket={bucket} onBookClick={onBookClick} />
        )}
      </Styled>
    )
  }
}
