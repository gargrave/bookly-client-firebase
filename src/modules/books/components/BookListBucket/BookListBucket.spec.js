import { bookBucketMocks } from '../../../../globals/mocks/'
import { ComponentBuilder } from '../../../../utils/testHelpers'

import BookBucketBookList from './BookBucketBookList/BookBucketBookList'
import BookBucketHeader from './BookBucketHeader/BookBucketHeader'

import BookListBucket from './BookListBucket'

const defaultProps = {
  bucket: bookBucketMocks[0],
  onBookClick: jest.fn(),
}

const builder = new ComponentBuilder(BookListBucket, defaultProps)

describe('BookListBucket', () => {
  let component

  beforeEach(() => {
    component = builder.shallowGetComponent()
  })

  describe('basic rendering', () => {
    it('matches the snapshot', () => {
      expect(component).toMatchSnapshot()
    })

    it('renders one BookBucketHeader', () => {
      expect(component.find(BookBucketHeader)).toHaveLength(1)
    })

    it('renders one BookBucketBookList when expanded', () => {
      component.instance().setState({ expanded: true })
      expect(component.find(BookBucketBookList)).toHaveLength(1)
    })

    it('renders zero BookBucketBookLists when collapsed', () => {
      component.instance().setState({ expanded: false })
      expect(component.find(BookBucketBookList)).toHaveLength(0)
    })
  })

  describe('state', () => {
    it('sets the correct default state', () => {
      const state = component.state()
      expect(state.expanded).toBe(true)
    })
  })

  describe('actions', () => {
    describe('handleToggleExpanded', () => {
      it('toggles the value of "expanded" in state', () => {
        let expanded = component.state().expanded
        component.instance().handleToggleExpanded()
        expect(component.state().expanded).toBe(!expanded)
        component.instance().handleToggleExpanded()
        expect(component.state().expanded).toBe(expanded)
      })
    })
  })
})
