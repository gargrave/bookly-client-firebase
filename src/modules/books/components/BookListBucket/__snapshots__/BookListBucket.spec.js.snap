// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookListBucket basic rendering matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BookListBucket
    bucket={
      Object {
        "author": "Aldous Huxley",
        "books": Array [
          Object {
            "author": Object {
              "bookCount": 1,
              "created": 2017-07-01T01:16:43.881Z,
              "firstName": "Aldous",
              "id": "oiw223h3qndlksadffl3ksdf",
              "lastName": "Huxley",
              "updated": 2017-07-01T01:16:43.881Z,
            },
            "created": 2018-01-14T01:36:26.220Z,
            "id": "sv3hmbefpc95MRXQsTHr",
            "sortBy": "",
            "title": "Brave New World",
            "updated": 2018-01-14T01:36:26.220Z,
          },
          Object {
            "author": Object {
              "bookCount": 1,
              "created": 2017-07-01T01:16:43.881Z,
              "firstName": "Aldous",
              "id": "oiw223h3qndlksadffl3ksdf",
              "lastName": "Huxley",
              "updated": 2017-07-01T01:16:43.881Z,
            },
            "created": 2018-01-15T01:36:26.220Z,
            "id": "Hmh9eR3rs5bMvsTXfpcQ",
            "sortBy": "",
            "title": "Island",
            "updated": 2018-01-15T01:36:26.220Z,
          },
          Object {
            "author": Object {
              "bookCount": 1,
              "created": 2017-07-01T01:16:43.881Z,
              "firstName": "Aldous",
              "id": "oiw223h3qndlksadffl3ksdf",
              "lastName": "Huxley",
              "updated": 2017-07-01T01:16:43.881Z,
            },
            "created": 2018-01-16T01:36:26.220Z,
            "id": "eRpcQ5H3rsTXfmh9bMvs",
            "sortBy": "",
            "title": "The Doors of Perception",
            "updated": 2018-01-16T01:36:26.220Z,
          },
        ],
      }
    }
    onBookClick={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <BookBucketHeader
          bucket={
            Object {
              "author": "Aldous Huxley",
              "books": Array [
                Object {
                  "author": Object {
                    "bookCount": 1,
                    "created": 2017-07-01T01:16:43.881Z,
                    "firstName": "Aldous",
                    "id": "oiw223h3qndlksadffl3ksdf",
                    "lastName": "Huxley",
                    "updated": 2017-07-01T01:16:43.881Z,
                  },
                  "created": 2018-01-14T01:36:26.220Z,
                  "id": "sv3hmbefpc95MRXQsTHr",
                  "sortBy": "",
                  "title": "Brave New World",
                  "updated": 2018-01-14T01:36:26.220Z,
                },
                Object {
                  "author": Object {
                    "bookCount": 1,
                    "created": 2017-07-01T01:16:43.881Z,
                    "firstName": "Aldous",
                    "id": "oiw223h3qndlksadffl3ksdf",
                    "lastName": "Huxley",
                    "updated": 2017-07-01T01:16:43.881Z,
                  },
                  "created": 2018-01-15T01:36:26.220Z,
                  "id": "Hmh9eR3rs5bMvsTXfpcQ",
                  "sortBy": "",
                  "title": "Island",
                  "updated": 2018-01-15T01:36:26.220Z,
                },
                Object {
                  "author": Object {
                    "bookCount": 1,
                    "created": 2017-07-01T01:16:43.881Z,
                    "firstName": "Aldous",
                    "id": "oiw223h3qndlksadffl3ksdf",
                    "lastName": "Huxley",
                    "updated": 2017-07-01T01:16:43.881Z,
                  },
                  "created": 2018-01-16T01:36:26.220Z,
                  "id": "eRpcQ5H3rsTXfmh9bMvs",
                  "sortBy": "",
                  "title": "The Doors of Perception",
                  "updated": 2018-01-16T01:36:26.220Z,
                },
              ],
            }
          }
          expanded={true}
          onToggleExpanded={[Function]}
        />,
        <BookBucketBookList
          bucket={
            Object {
              "author": "Aldous Huxley",
              "books": Array [
                Object {
                  "author": Object {
                    "bookCount": 1,
                    "created": 2017-07-01T01:16:43.881Z,
                    "firstName": "Aldous",
                    "id": "oiw223h3qndlksadffl3ksdf",
                    "lastName": "Huxley",
                    "updated": 2017-07-01T01:16:43.881Z,
                  },
                  "created": 2018-01-14T01:36:26.220Z,
                  "id": "sv3hmbefpc95MRXQsTHr",
                  "sortBy": "",
                  "title": "Brave New World",
                  "updated": 2018-01-14T01:36:26.220Z,
                },
                Object {
                  "author": Object {
                    "bookCount": 1,
                    "created": 2017-07-01T01:16:43.881Z,
                    "firstName": "Aldous",
                    "id": "oiw223h3qndlksadffl3ksdf",
                    "lastName": "Huxley",
                    "updated": 2017-07-01T01:16:43.881Z,
                  },
                  "created": 2018-01-15T01:36:26.220Z,
                  "id": "Hmh9eR3rs5bMvsTXfpcQ",
                  "sortBy": "",
                  "title": "Island",
                  "updated": 2018-01-15T01:36:26.220Z,
                },
                Object {
                  "author": Object {
                    "bookCount": 1,
                    "created": 2017-07-01T01:16:43.881Z,
                    "firstName": "Aldous",
                    "id": "oiw223h3qndlksadffl3ksdf",
                    "lastName": "Huxley",
                    "updated": 2017-07-01T01:16:43.881Z,
                  },
                  "created": 2018-01-16T01:36:26.220Z,
                  "id": "eRpcQ5H3rsTXfmh9bMvs",
                  "sortBy": "",
                  "title": "The Doors of Perception",
                  "updated": 2018-01-16T01:36:26.220Z,
                },
              ],
            }
          }
          onBookClick={[Function]}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "bucket": Object {
            "author": "Aldous Huxley",
            "books": Array [
              Object {
                "author": Object {
                  "bookCount": 1,
                  "created": 2017-07-01T01:16:43.881Z,
                  "firstName": "Aldous",
                  "id": "oiw223h3qndlksadffl3ksdf",
                  "lastName": "Huxley",
                  "updated": 2017-07-01T01:16:43.881Z,
                },
                "created": 2018-01-14T01:36:26.220Z,
                "id": "sv3hmbefpc95MRXQsTHr",
                "sortBy": "",
                "title": "Brave New World",
                "updated": 2018-01-14T01:36:26.220Z,
              },
              Object {
                "author": Object {
                  "bookCount": 1,
                  "created": 2017-07-01T01:16:43.881Z,
                  "firstName": "Aldous",
                  "id": "oiw223h3qndlksadffl3ksdf",
                  "lastName": "Huxley",
                  "updated": 2017-07-01T01:16:43.881Z,
                },
                "created": 2018-01-15T01:36:26.220Z,
                "id": "Hmh9eR3rs5bMvsTXfpcQ",
                "sortBy": "",
                "title": "Island",
                "updated": 2018-01-15T01:36:26.220Z,
              },
              Object {
                "author": Object {
                  "bookCount": 1,
                  "created": 2017-07-01T01:16:43.881Z,
                  "firstName": "Aldous",
                  "id": "oiw223h3qndlksadffl3ksdf",
                  "lastName": "Huxley",
                  "updated": 2017-07-01T01:16:43.881Z,
                },
                "created": 2018-01-16T01:36:26.220Z,
                "id": "eRpcQ5H3rsTXfmh9bMvs",
                "sortBy": "",
                "title": "The Doors of Perception",
                "updated": 2018-01-16T01:36:26.220Z,
              },
            ],
          },
          "expanded": true,
          "onToggleExpanded": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "bucket": Object {
            "author": "Aldous Huxley",
            "books": Array [
              Object {
                "author": Object {
                  "bookCount": 1,
                  "created": 2017-07-01T01:16:43.881Z,
                  "firstName": "Aldous",
                  "id": "oiw223h3qndlksadffl3ksdf",
                  "lastName": "Huxley",
                  "updated": 2017-07-01T01:16:43.881Z,
                },
                "created": 2018-01-14T01:36:26.220Z,
                "id": "sv3hmbefpc95MRXQsTHr",
                "sortBy": "",
                "title": "Brave New World",
                "updated": 2018-01-14T01:36:26.220Z,
              },
              Object {
                "author": Object {
                  "bookCount": 1,
                  "created": 2017-07-01T01:16:43.881Z,
                  "firstName": "Aldous",
                  "id": "oiw223h3qndlksadffl3ksdf",
                  "lastName": "Huxley",
                  "updated": 2017-07-01T01:16:43.881Z,
                },
                "created": 2018-01-15T01:36:26.220Z,
                "id": "Hmh9eR3rs5bMvsTXfpcQ",
                "sortBy": "",
                "title": "Island",
                "updated": 2018-01-15T01:36:26.220Z,
              },
              Object {
                "author": Object {
                  "bookCount": 1,
                  "created": 2017-07-01T01:16:43.881Z,
                  "firstName": "Aldous",
                  "id": "oiw223h3qndlksadffl3ksdf",
                  "lastName": "Huxley",
                  "updated": 2017-07-01T01:16:43.881Z,
                },
                "created": 2018-01-16T01:36:26.220Z,
                "id": "eRpcQ5H3rsTXfmh9bMvs",
                "sortBy": "",
                "title": "The Doors of Perception",
                "updated": 2018-01-16T01:36:26.220Z,
              },
            ],
          },
          "onBookClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <BookBucketHeader
            bucket={
              Object {
                "author": "Aldous Huxley",
                "books": Array [
                  Object {
                    "author": Object {
                      "bookCount": 1,
                      "created": 2017-07-01T01:16:43.881Z,
                      "firstName": "Aldous",
                      "id": "oiw223h3qndlksadffl3ksdf",
                      "lastName": "Huxley",
                      "updated": 2017-07-01T01:16:43.881Z,
                    },
                    "created": 2018-01-14T01:36:26.220Z,
                    "id": "sv3hmbefpc95MRXQsTHr",
                    "sortBy": "",
                    "title": "Brave New World",
                    "updated": 2018-01-14T01:36:26.220Z,
                  },
                  Object {
                    "author": Object {
                      "bookCount": 1,
                      "created": 2017-07-01T01:16:43.881Z,
                      "firstName": "Aldous",
                      "id": "oiw223h3qndlksadffl3ksdf",
                      "lastName": "Huxley",
                      "updated": 2017-07-01T01:16:43.881Z,
                    },
                    "created": 2018-01-15T01:36:26.220Z,
                    "id": "Hmh9eR3rs5bMvsTXfpcQ",
                    "sortBy": "",
                    "title": "Island",
                    "updated": 2018-01-15T01:36:26.220Z,
                  },
                  Object {
                    "author": Object {
                      "bookCount": 1,
                      "created": 2017-07-01T01:16:43.881Z,
                      "firstName": "Aldous",
                      "id": "oiw223h3qndlksadffl3ksdf",
                      "lastName": "Huxley",
                      "updated": 2017-07-01T01:16:43.881Z,
                    },
                    "created": 2018-01-16T01:36:26.220Z,
                    "id": "eRpcQ5H3rsTXfmh9bMvs",
                    "sortBy": "",
                    "title": "The Doors of Perception",
                    "updated": 2018-01-16T01:36:26.220Z,
                  },
                ],
              }
            }
            expanded={true}
            onToggleExpanded={[Function]}
          />,
          <BookBucketBookList
            bucket={
              Object {
                "author": "Aldous Huxley",
                "books": Array [
                  Object {
                    "author": Object {
                      "bookCount": 1,
                      "created": 2017-07-01T01:16:43.881Z,
                      "firstName": "Aldous",
                      "id": "oiw223h3qndlksadffl3ksdf",
                      "lastName": "Huxley",
                      "updated": 2017-07-01T01:16:43.881Z,
                    },
                    "created": 2018-01-14T01:36:26.220Z,
                    "id": "sv3hmbefpc95MRXQsTHr",
                    "sortBy": "",
                    "title": "Brave New World",
                    "updated": 2018-01-14T01:36:26.220Z,
                  },
                  Object {
                    "author": Object {
                      "bookCount": 1,
                      "created": 2017-07-01T01:16:43.881Z,
                      "firstName": "Aldous",
                      "id": "oiw223h3qndlksadffl3ksdf",
                      "lastName": "Huxley",
                      "updated": 2017-07-01T01:16:43.881Z,
                    },
                    "created": 2018-01-15T01:36:26.220Z,
                    "id": "Hmh9eR3rs5bMvsTXfpcQ",
                    "sortBy": "",
                    "title": "Island",
                    "updated": 2018-01-15T01:36:26.220Z,
                  },
                  Object {
                    "author": Object {
                      "bookCount": 1,
                      "created": 2017-07-01T01:16:43.881Z,
                      "firstName": "Aldous",
                      "id": "oiw223h3qndlksadffl3ksdf",
                      "lastName": "Huxley",
                      "updated": 2017-07-01T01:16:43.881Z,
                    },
                    "created": 2018-01-16T01:36:26.220Z,
                    "id": "eRpcQ5H3rsTXfmh9bMvs",
                    "sortBy": "",
                    "title": "The Doors of Perception",
                    "updated": 2018-01-16T01:36:26.220Z,
                  },
                ],
              }
            }
            onBookClick={[Function]}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "bucket": Object {
              "author": "Aldous Huxley",
              "books": Array [
                Object {
                  "author": Object {
                    "bookCount": 1,
                    "created": 2017-07-01T01:16:43.881Z,
                    "firstName": "Aldous",
                    "id": "oiw223h3qndlksadffl3ksdf",
                    "lastName": "Huxley",
                    "updated": 2017-07-01T01:16:43.881Z,
                  },
                  "created": 2018-01-14T01:36:26.220Z,
                  "id": "sv3hmbefpc95MRXQsTHr",
                  "sortBy": "",
                  "title": "Brave New World",
                  "updated": 2018-01-14T01:36:26.220Z,
                },
                Object {
                  "author": Object {
                    "bookCount": 1,
                    "created": 2017-07-01T01:16:43.881Z,
                    "firstName": "Aldous",
                    "id": "oiw223h3qndlksadffl3ksdf",
                    "lastName": "Huxley",
                    "updated": 2017-07-01T01:16:43.881Z,
                  },
                  "created": 2018-01-15T01:36:26.220Z,
                  "id": "Hmh9eR3rs5bMvsTXfpcQ",
                  "sortBy": "",
                  "title": "Island",
                  "updated": 2018-01-15T01:36:26.220Z,
                },
                Object {
                  "author": Object {
                    "bookCount": 1,
                    "created": 2017-07-01T01:16:43.881Z,
                    "firstName": "Aldous",
                    "id": "oiw223h3qndlksadffl3ksdf",
                    "lastName": "Huxley",
                    "updated": 2017-07-01T01:16:43.881Z,
                  },
                  "created": 2018-01-16T01:36:26.220Z,
                  "id": "eRpcQ5H3rsTXfmh9bMvs",
                  "sortBy": "",
                  "title": "The Doors of Perception",
                  "updated": 2018-01-16T01:36:26.220Z,
                },
              ],
            },
            "expanded": true,
            "onToggleExpanded": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "bucket": Object {
              "author": "Aldous Huxley",
              "books": Array [
                Object {
                  "author": Object {
                    "bookCount": 1,
                    "created": 2017-07-01T01:16:43.881Z,
                    "firstName": "Aldous",
                    "id": "oiw223h3qndlksadffl3ksdf",
                    "lastName": "Huxley",
                    "updated": 2017-07-01T01:16:43.881Z,
                  },
                  "created": 2018-01-14T01:36:26.220Z,
                  "id": "sv3hmbefpc95MRXQsTHr",
                  "sortBy": "",
                  "title": "Brave New World",
                  "updated": 2018-01-14T01:36:26.220Z,
                },
                Object {
                  "author": Object {
                    "bookCount": 1,
                    "created": 2017-07-01T01:16:43.881Z,
                    "firstName": "Aldous",
                    "id": "oiw223h3qndlksadffl3ksdf",
                    "lastName": "Huxley",
                    "updated": 2017-07-01T01:16:43.881Z,
                  },
                  "created": 2018-01-15T01:36:26.220Z,
                  "id": "Hmh9eR3rs5bMvsTXfpcQ",
                  "sortBy": "",
                  "title": "Island",
                  "updated": 2018-01-15T01:36:26.220Z,
                },
                Object {
                  "author": Object {
                    "bookCount": 1,
                    "created": 2017-07-01T01:16:43.881Z,
                    "firstName": "Aldous",
                    "id": "oiw223h3qndlksadffl3ksdf",
                    "lastName": "Huxley",
                    "updated": 2017-07-01T01:16:43.881Z,
                  },
                  "created": 2018-01-16T01:36:26.220Z,
                  "id": "eRpcQ5H3rsTXfmh9bMvs",
                  "sortBy": "",
                  "title": "The Doors of Perception",
                  "updated": 2018-01-16T01:36:26.220Z,
                },
              ],
            },
            "onBookClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
